src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs

```diff
@@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor
         /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>
         public TriggerDiagnosticDescriptor(string id, params string[] customTags)
             : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,
-                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))
+                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))
```

Comment:
What I am saying is if CustomTag.Trigger is set, make sure SimpleDiagnostic.Create to internally create TriggerDescriptor and drop all other fields not needed to TriggerDescriptor.


-          heejae

From: Shyam N [mailto:notifications@github.com]
Sent: Tuesday, February 24, 2015 4:11 PM
To: dotnet/roslyn
Cc: HeeJae Chang
Subject: Re: [roslyn] Introduce a new well known custom tag for trigger diagnostic. (#815)


In src/Compilers/Core/Portable/Diagnostic/TriggerDiagnosticDescriptor.cs<https://github.com/dotnet/roslyn/pull/815#discussion_r25307184>:

> @@ -18,7 +18,7 @@ public class TriggerDiagnosticDescriptor : DiagnosticDescriptor

>          /// <param name="customTags">Optional custom tags for the diagnostic. See <see cref="WellKnownDiagnosticTags"/> for some well known tags.</param>

>          public TriggerDiagnosticDescriptor(string id, params string[] customTags)

>              : base(id, title: "", messageFormat: "", category: "", defaultSeverity: DiagnosticSeverity.Hidden, isEnabledByDefault: true,

> -                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable))

> +                    customTags: customTags.Append(WellKnownDiagnosticTags.NotConfigurable).Append(WellKnownDiagnosticTags.Trigger))

Not sure I understand. SimpleDiagnostic.Create allows specifying all fields for a descriptor. TriggerDiagnosticDescriptor only supports id and tags (since trigger diagnostics are never displayed to user anywhere). Are you saying that TriggerDiagnosticDescriptor should be expanded to allow all descriptor fields?

—
Reply to this email directly or view it on GitHub<https://github.com/dotnet/roslyn/pull/815/files#r25307184>.